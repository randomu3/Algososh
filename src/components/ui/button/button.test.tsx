// link.test.js
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { Button } from "./button";

describe("Проверяем при помощи снэпшотов корректную отрисовку:", () => {
  it("-> Кнопки с текстом", () => {
    render(<Button text={"text"} />);
    const button = screen.getByRole("button");
    expect(button).toMatchSnapshot();
  });

  it("-> Кнопки без текста", () => {
    render(<Button text={""} />);
    const button = screen.getByRole("button");
    expect(button).toMatchSnapshot();
  });

  it("-> Заблокированной кнопки", () => {
    render(<Button disabled />);
    const button = screen.getByRole<HTMLButtonElement>("button");
    expect(button).toMatchSnapshot();
  });

  it("-> Кнопки с индикацией загрузки", () => {
    render(<Button isLoader={true} />);
    const button = screen.getByRole("button");
    expect(button).toMatchSnapshot();
  });

  it("-> Вызова колбека при клике на кнопку", () => {
    const mockFn = jest.fn();
    render(<Button onClick={mockFn}/>);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(mockFn).toHaveBeenCalled();
  });
});
