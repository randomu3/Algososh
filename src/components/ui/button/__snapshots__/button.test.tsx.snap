// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяем при помощи снэпшотов корректную отрисовку: -> Заблокированной кнопки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку: -> Кнопки без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку: -> Кнопки с индикацией загрузки 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку: -> Кнопки с текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    text
  </p>
</button>
`;
